categories/99-problems/P46-vaibhav.pl

```diff
@@ -0,0 +1,87 @@
+use v6;
+
+=begin pod
+
+= TITLE P46 - Truth tables for logical expressions
+
+= AUTHOR Vaibhav Mehra
+
+=head1 Example:
+
+    > my $result = and($A, or($A, not($B)));
+    true true true
+    true fail true
+    fail true fail
+    fail fail fail
+
+=end pod
+
+# we'll define subroutines
+# for all the basic logic gates.
+
+multi and ($input1, $input2) {
+  if $input1 eq 'true' && $input2 eq 'true' {
+    return 'true';
+  }
+  else {
+    return 'fail';
+  }
+}
+
+multi or ($input1, $input2) {
+  if $input1 eq 'fail' && $input2 eq 'fail' {
+    return 'fail';
+  }
+  else {
+    return 'true';
+  }
+}
+
+sub not ($input) {
+  if $input eq 'true' {
+    return 'fail';
+  }
+  else {
+    return 'true';
+  }
+}
+
+multi nor ($input1, $input2) {
+  return not(or($input1, $input2));
+}
+
+multi nand ($input1, $input2) {
+  return not(and($input1, $input2));
+}
+
+multi xor ($input1, $input2) {
+  return or(and($input1, not($input2)), and(not($input1), $input2));
+}
+
+multi xnor ($input1, $input2) {
+  return and(or($input1, not($input2)), or(not($input1), $input2));
+}
+
+# Using an example to understand the working
+my $A = 'true';
+my $B = 'true';
+my $result = and($A, or($A, not($B)));
+print "$A ";
+print "$B ";
+say $result;
+
+$B = 'fail';
+print "$A ";
+print "$B ";
+say $result;
```

Comment:
You are not computing $result. It will have the same value as before.

Reply:
Sir should I have to create another pull request to commit required changes or please guide me how to create changes in this one? Sorry for inconvenience from my side @JJ

Reply:
You just have to push the changes to your original branch and they'll show up here.